UAlbertaBot_Config.txt
{
    "Bot Info" :
    {
        "BotName" : "UAlbertaBot",
        "Authors" : "David Churchill",
		"Race"    : "Zerg",
        "PrintInfoOnStart" : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"   : 10,
        "SetFrameSkip"    : 0,
        "EnableUserInput" : true,
        "EnableCompleteMapInformation" : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation" : true,
    
        "WorkerDefense" : true,
        "WorkerDefensePerUnit" : 2,
        
        "InCombatRadius" : 1000,
        "RegroupRadius" : 300,
        "UnitNearEnemyRadius" : 600
    },
    
    "Macro" :
    {
        "BuildingSpacing" : 1,
        "PylonSpacing"   : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "bwapi-data/AI/UAlbertaBot_ErrorLog.txt",
        
        "PrintModuleTimeout"        : false,
        "DrawBuildOrderSearchInfo"  : true,
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true,
        "DrawScoutInfo"             : false,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        
        "AllDebugOff"               : false
    },
    
    "Modules" :
    {
        "UseGameCommander"      : true,
        "UseScoutManager"       : true,
        "UseCombatCommander"    : true,
        "UseBuildOrderSearch"   : true,
        "UseStrategyIO"         : true,
        "UseAutoObserver"       : false
    },
    
    "Tools" :
    {
        "MapGridSize" : 320
    },
    
    "Strategy" :
    {
        "ProtossStrategyName"   : "Protoss_ZealotRush",
        "TerranStrategyName"    : "Terran_MarineRush",
        "ZergStrategyName"      : "",
        
        "ScoutGasSteal"         : false,
        "ScoutHarassEnemy"      : true,
        
        "ReadDirectory"         : "bwapi-data/read/",
        "WriteDirectory"        : "bwapi-data/write/",
        
        "Strategies" :
        {
            "Protoss_ZealotRush"    : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Gateway", "Probe", "Probe", "Zealot", "Pylon", "Zealot", "Zealot",  "Probe", "Zealot", "Zealot", "Probe", "Pylon", "Zealot", "Gateway", "Probe", "Pylon", "Probe", "Zealot", "Probe", "Zealot", "Zealot", "Zealot", "Zealot", "Pylon", "Probe", "Zealot", "Zealot", "Zealot" ]},
            "Protoss_DragoonRush"   : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Probe", "Cybernetics_Core", "Probe", "Probe", "Gateway", "Singularity_Charge", "Dragoon", "Gateway", "Pylon", "Dragoon", "Dragoon", "Probe", "Gateway", "Pylon", "Probe", "Dragoon", "Dragoon", "Dragoon" ]},
            "Protoss_DTRush"        : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Cybernetics_Core", "Probe", "Citadel_of_Adun", "Probe", "Templar_Archives", "Gateway", "Dark_Templar", "Dark_Templar", "Pylon", "Dark_Templar", "Dark_Templar", "Probe", "Pylon", "Probe" ]},
            "Terran_MarineRush"     : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "SCV"]},
            "Terran_VultureRush"    : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "SCV", "SCV", "Barracks", "Refinery", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "Factory", "SCV", "SCV", "SCV", "Vulture"]},
            "Zerg_9Pool"     		: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone","Spawning Pool","Drone","Extractor","Overlord","Drone","Zergling","Zergling","Zergling","Metabolic_Boost","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Hatchery"] },
            "Zerg_3HatchMuta"       : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Hatchery", "Extractor", "Drone", "Drone", "Drone", "Overlord", "Lair", "Extractor", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Overlord", "Drone", "Overlord", "Spire", "Drone", "Drone", "Drone", "Drone", "Drone", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk"]},
			"Zerg_9/10Hatch"     	: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor","Drone","Hatchery", "Overlord", "Drone", "Spawning_Pool",  "Drone", "Drone", "Drone","Drone","Zergling", "Zergling", "Zergling", "Zergling", "Drone", "Drone", "Overlord", "Hatchery", "Drone", "Drone", "Drone", "Drone", "Drone"] },
						"Zerg_9/10Hatch_ne"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Creep_Colony", "Drone","Drone","Hatchery","Drone", "Drone", "Drone","Drone", "Creep_Colony", "Extractor" ,"Drone" ,"Drone" ,"Drone" , "Creep_Colony","Drone","Zergling","Drone", "Zergling", "Hydralisk Den", "Extractor", "Drone", "Extractor", "Creep_Colony", "Overlord", "Overlord","Overlord"]},
			"Zerg_9/10Hatch_ep"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Zergling", "Zergling", "Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling", "Hydralisk_Den", "Drone" ,"Drone" ,"Drone" ,"Drone" , "Overlord","Overlord","Overlord","Muscular_Augments"] },
			"Zerg_3HatchHydra"    	: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning_Pool", "Drone", "Drone", "Drone", "Hatchery", "Zergling", "Zergling", "Zergling", "Zergling","Drone", "Hatchery", "Drone", "Overlord", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Drone"] },
			"Zerg_3HatchHydra_ne"   : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Drone", "Drone", "Drone", "Lair", "Drone", "Extractor", "Overlord", "Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone",  "Drone", "Spire", "Drone", "Drone", "Drone", "Hatchery", "Creep_Colony", "Metabolic_Boost", "Drone", "Drone", "Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Overlord",  "Zerg_Flyer_Carapace", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk" ] },
			"Zerg_3HatchHydra_ep"   : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Overlord", "Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Drone", "Lair", "Drone", "Creep_Colony", "Drone", "Drone", "Creep_Colony", "Hatchery", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Spire", "Drone", "Drone", "Drone", "Drone",  "Drone", "Drone", "Drone", "Creep_Colony", "Overlord", "Overlord", "Overlord","Zerg_Flyer_Carapace","Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk"  ] }
        }
    }
}

UAlbertaBot_ConfigOLD.txt
{
    "Bot Info" :
    {
        "BotName" : "UAlbertaBot",
        "Authors" : "David Churchill",
		"Race"    : "Zerg",
        "PrintInfoOnStart" : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"   : 10,
        "SetFrameSkip"    : 0,
        "EnableUserInput" : true,
        "EnableCompleteMapInformation" : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation" : true,
    
        "WorkerDefense" : true,
        "WorkerDefensePerUnit" : 2,
        
        "InCombatRadius" : 1000,
        "RegroupRadius" : 300,
        "UnitNearEnemyRadius" : 600
    },
    
    "Macro" :
    {
        "BuildingSpacing" : 1,
        "PylonSpacing"   : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "bwapi-data/AI/UAlbertaBot_ErrorLog.txt",
        
        "PrintModuleTimeout"        : false,
        "DrawBuildOrderSearchInfo"  : true,
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true,
        "DrawScoutInfo"             : false,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        
        "AllDebugOff"               : false
    },
    
    "Modules" :
    {
        "UseGameCommander"      : true,
        "UseScoutManager"       : true,
        "UseCombatCommander"    : true,
        "UseBuildOrderSearch"   : true,
        "UseStrategyIO"         : true,
        "UseAutoObserver"       : false
    },
    
    "Tools" :
    {
        "MapGridSize" : 320
    },
    
    "Strategy" :
    {
        "ProtossStrategyName"   : "Protoss_ZealotRush",
        "TerranStrategyName"    : "Terran_MarineRush",
        "ZergStrategyName"      : "",
        
        "ScoutGasSteal"         : false,
        "ScoutHarassEnemy"      : true,
        
        "ReadDirectory"         : "bwapi-data/read/",
        "WriteDirectory"        : "bwapi-data/write/",
        
        "Strategies" :
        {
            "Protoss_ZealotRush"    : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Gateway", "Probe", "Probe", "Zealot", "Pylon", "Zealot", "Zealot",  "Probe", "Zealot", "Zealot", "Probe", "Pylon", "Zealot", "Gateway", "Probe", "Pylon", "Probe", "Zealot", "Probe", "Zealot", "Zealot", "Zealot", "Zealot", "Pylon", "Probe", "Zealot", "Zealot", "Zealot" ]},
            "Protoss_DragoonRush"   : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Probe", "Cybernetics_Core", "Probe", "Probe", "Gateway", "Singularity_Charge", "Dragoon", "Gateway", "Pylon", "Dragoon", "Dragoon", "Probe", "Gateway", "Pylon", "Probe", "Dragoon", "Dragoon", "Dragoon" ]},
            "Protoss_DTRush"        : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Cybernetics_Core", "Probe", "Citadel_of_Adun", "Probe", "Templar_Archives", "Gateway", "Dark_Templar", "Dark_Templar", "Pylon", "Dark_Templar", "Dark_Templar", "Probe", "Pylon", "Probe" ]},
            "Terran_MarineRush"     : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "SCV"]},
            "Terran_VultureRush"    : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "SCV", "SCV", "Barracks", "Refinery", "SCV", "SCV", "SCV", "SCV", "Supply Depot", "Factory", "SCV", "SCV", "SCV", "Vulture"]},
            "Zerg_9Pool"     		: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone","Spawning Pool","Drone","Extractor","Overlord","Drone","Zergling","Zergling","Zergling","Metabolic_Boost","Zergling","Zergling","Zergling","Hatchery","Zergling","Zergling","Zergling","Hatchery"] },
            "Zerg_3HatchMuta"       : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Drone", "Drone", "Drone", "Hatchery", "Spawning_Pool", "Drone", "Drone", "Hatchery", "Extractor", "Drone", "Drone", "Drone", "Overlord", "Lair", "Extractor", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Drone", "Drone", "Drone", "Drone", "Overlord", "Drone", "Overlord", "Drone", "Overlord", "Spire", "Drone", "Drone", "Drone", "Drone", "Drone", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk"]},
			"Zerg_9/10Hatch"     	: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor","Drone","Hatchery", "Overlord", "Drone", "Spawning_Pool",  "Drone", "Drone", "Drone","Drone","Zergling", "Zergling", "Zergling", "Zergling", "Drone", "Drone", "Overlord", "Hatchery", "Drone", "Drone", "Drone", "Drone", "Drone"] },
						"Zerg_9/10Hatch_ne"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Creep_Colony", "Drone","Drone","Hatchery","Drone", "Drone", "Drone","Drone", "Creep_Colony", "Extractor" ,"Drone" ,"Drone" ,"Drone" , "Creep_Colony","Drone","Zergling","Drone", "Zergling", "Hydralisk Den", "Extractor", "Drone", "Extractor", "Creep_Colony", "Overlord", "Overlord","Overlord"]},
			"Zerg_9/10Hatch_ep"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Zergling", "Zergling", "Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling","Zergling", "Hydralisk_Den", "Drone" ,"Drone" ,"Drone" ,"Drone" , "Overlord","Overlord","Overlord","Muscular_Augments"] },
			"Zerg_3HatchHydra"    	: { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning_Pool", "Drone", "Drone", "Drone", "Hatchery", "Zergling", "Zergling", "Zergling", "Zergling","Drone", "Hatchery", "Drone", "Overlord", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Drone"] },
			"Zerg_3HatchHydra_ne"   : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Drone", "Drone", "Drone", "Lair", "Drone", "Extractor", "Overlord", "Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone", "Drone",  "Drone", "Drone", "Spire", "Drone", "Drone", "Drone", "Hatchery", "Creep_Colony", "Metabolic_Boost", "Drone", "Drone", "Drone", "Drone", "Creep_Colony", "Drone", "Drone", "Overlord", "Flyer_Carapace", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk", "Mutalisk" ] },
			"Zerg_3HatchHydra_ep"   : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Hatchery", "Drone", "Drone", "Drone", "Creep_Colony", "Creep_Colony", "Drone", "Drone", "Drone", "Extractor", "Extractor", "Drone", "Drone", "Creep_Colony", "Creep_Colony", "Overlord", "Evolution_Chamber", "Evolution_Chamber", "Drone", "Drone", "Drone", "Drone", "Overlord",  "Hydralisk_Den", "Overlord", "Overlord", "Overlord", "Muscular_Augments"] }
        }
    }
}

MergedWithDingkai/UAlbertaBot/Source/Config.cpp
#include "Config.h"
#include "UABAssert.h"

namespace Config
{
    namespace ConfigFile
    {
        bool ConfigFileFound                = false;
        bool ConfigFileParsed               = false;
		std::string ConfigFileLocation = "C:/Users/Soul/Documents/School/CMPUT/350/DingkaiV3/c350-DingkaiZVZ/c350-Dingkai/UAlbertaBot/UAlbertaBot_Config.txt";
    }

    namespace Strategy
    {
        std::string AgainstProtossStrategyName     = "Zerg_9/10Hatch";
        std::string AgainstTerrenStrategyName      = "Zerg_3HatchHydra";
        std::string AgainstZergStrategyName        = "Zerg_9Pool";
		std::string StrategyName			= "";
        std::string ReadDir                 = "bwapi-data/read/";
        std::string WriteDir                = "bwapi-data/write/";
        bool GasStealWithScout              = false;
        bool ScoutHarassEnemy               = true;
        bool UseEnemySpecificStrategy       = false;
        bool FoundEnemySpecificStrategy     = false;
    }

    namespace Modules							    
    {
        // the default tournament bot modules
        bool UsingGameCommander             = true;     // toggle GameCommander, effectively UAlbertaBot
        bool UsingCombatCommander           = true;
        bool UsingBuildOrderSearch          = true;     // toggle use of Build Order Search, currently no backup
        bool UsingAutoObserver              = false;
        bool UsingStrategyIO                = false;    // toggle the use of file io for strategy
        bool UsingUnitCommandManager        = false;    // handles all unit commands
		
        // extra things, don't enable unless you know what they are
        bool UsingBuildOrderDemo            = false;
    }

    namespace BotInfo
    {
        std::string BotName                 = "UAlbertaBot";
        std::string Authors                 = "Dave Churchill";
        bool PrintInfoOnStart               = false;
    }

    namespace BWAPIOptions
    {
        int SetLocalSpeed                   = 42;
        int SetFrameSkip                    = 0;
        bool EnableUserInput                = true;
        bool EnableCompleteMapInformation   = false;
    }
    
    namespace Tournament						
    {
        int GameEndFrame                    = 86400;	
    }
    
    namespace Debug								
    {
        bool DrawGameInfo                   = true;
        bool DrawUnitHealthBars             = true;
        bool DrawProductionInfo             = true;
        bool DrawBuildOrderSearchInfo       = true;
        bool DrawScoutInfo                  = false;
        bool DrawResourceInfo               = false;
        bool DrawWorkerInfo                 = false;
        bool DrawModuleTimers               = false;
        bool DrawReservedBuildingTiles      = false;
        bool DrawCombatSimulationInfo       = false;
        bool DrawBuildingInfo               = false;
        bool DrawMouseCursorInfo            = false;
        bool DrawEnemyUnitInfo              = false;
        bool DrawBWTAInfo                   = false;
        bool DrawMapGrid                    = false;
        bool DrawUnitTargetInfo             = false;
        bool DrawSquadInfo                  = false;
        bool DrawBOSSStateInfo              = false;
        bool PrintModuleTimeout             = false;	

        std::string ErrorLogFilename        = "UAB_ErrorLog.txt";
        bool LogAssertToErrorFile           = false;

        BWAPI::Color ColorLineTarget        = BWAPI::Colors::White;
        BWAPI::Color ColorLineMineral       = BWAPI::Colors::Cyan;
        BWAPI::Color ColorUnitNearEnemy     = BWAPI::Colors::Red;
        BWAPI::Color ColorUnitNotNearEnemy  = BWAPI::Colors::Green;
    }

    namespace Micro								
    {
        bool UseSparcraftSimulation         = true;
        bool KiteWithRangedUnits            = true;
        std::set<BWAPI::UnitType> KiteLongerRangedUnits;
		bool WorkersDefendRush				= false;
		int RetreatMeleeUnitShields         = 0;
        int RetreatMeleeUnitHP              = 0;
        int CombatRadius                    = 1000;     // radius of combat to consider units for Micro Search
        int CombatRegroupRadius             = 300;      // radius of units around frontmost unit we consider in regroup calculation
        int UnitNearEnemyRadius             = 600;      // radius to consider a unit 'near' to an enemy unit
    }

    namespace Macro
    {
        int BOSSFrameLimit                  = 160;
        int WorkersPerRefinery              = 3;
        int BuildingSpacing                 = 1;
        int PylonSpacing                    = 3;
    }

    namespace Tools								
    {
        extern int MAP_GRID_SIZE            = 320;      // size of grid spacing in MapGrid
    }
}